table/table.cpp

```diff
@@ -164,59 +133,120 @@ void Table::edit_rows(vector<string> &columns,
             }
             ::database::Entry* edit_entry = table.mutable_rows(i)->mutable_entries(edit_indecies[j].index);
             edit_entry->CopyFrom(entries[j]);
         }
     }
 } // edit_rows()
 
-void Table::insert(::database::Row &row) {
-    for(auto it : column_indecies) {
+// Edits all rows to set entry@each edit_column = the said new entry
+// Input: vector<string> &edit_columns -> columns being edited in filtered entries
+// Input: vector<::database::Entry> &entries -> new entries to update entries in edit_columns
+void Table::edit_all(vector<string> &edit_columns, 
+                vector<::database::Entry> &entries) {
+    vector<Info> edit_indecies;
+    for (string column: edit_columns) {
         try {
-            if(!correct_type(it.second, row.entries(it.second.index))) {
-                throw out_of_range("Row is wrong type");
+            Info index = column_index(column);
+            edit_indecies.push_back(index);
+        } catch (std::exception e) {
+            throw e;
+        }
+    }
+
+    for(size_t i = 0; i < table.rows_size(); ++i) {
+        // Edits each specified entry in row
+        for(size_t j = 0; j < entries.size(); ++j) {
+            if(!correct_type(edit_indecies[j], entries[j])) {
+                throw type_mismatch(type_mismatch::error_type::updation, get_type(edit_indecies[j]), get_type(entries[j]));
             }
-        } catch (exception e) {
-            throw out_of_range("Row is wrong type");
+            ::database::Entry* edit_entry = table.mutable_rows(i)->mutable_entries(edit_indecies[j].index);
+            edit_entry->CopyFrom(entries[j]);
+        }
+    }
+}
+
+// Inserts row into database
+// Checks if row aligns with correct types
+// Input: ::database::row &row -> row being inserted
+void Table::insert(::database::Row &row) {
+    for(auto it : column_indecies) {
+        if(it.second.index > row.entries_size() - 1) {
+            stringstream error_message;
+            error_message << "ERROR: Row being inserted is too small! Row has " << row.entries_size() << " entries, when it is supposed to have " << column_indecies.size();
+            throw out_of_range(error_message.str());
+        } else if(!correct_type(it.second, row.entries(it.second.index))) {
+            cout << "Throwing type mismatch" << endl;
+            throw type_mismatch(type_mismatch::error_type::insertion, get_type(row.entries(it.second.index)), get_type(it.second));
         }
-        
     }
     ::database::Row* new_row = table.add_rows();
     *new_row = row;
 } // insert()
 
+// Removes rows where each entry@each column = the said comparison
+// Input: vector<string> columns -> columns being accessed
+// Input: vector<Entry> comparisons -> list of entries to compare each row entry@column to
+void Table::remove_rows(vector<string> &columns, vector<::database::Entry> &comparisons) {
+    vector<Info> indecies;
+    
+    for (string column: columns) {
```

Comment:
nit: Same thing as above, please add a space between the variable and the colon.

i.e.
`for (string column : columns) {`